vespa/application.py

```diff
@@ -414,7 +414,7 @@ def _submit(doc:dict, sync_session:VespaSync) -> Tuple[str, Union[VespaResponse,
             except Exception as e:
                 return (id, e)
 
-        def _handle_result_callback(future:Future, callback:Callable):
+        def _handle_result_callback(future: Future, callback: Optional[Callable[[VespaResponse, str], None]]):
```

Comment:
We can assume that the type of the callback function is `Callable[[VespaResponse, str], None]` as documented

```python
:param callback: A callback function to be called on each result. Signature `callback(response:VespaResponse, id:str)`
```

Also, it's optional.